### PR TITLE
Adding a model class for fabric8 gemini server

### PR DESCRIPTION
**Pull request by GeetikaBatra**

### CODE REVIEW

**Comment by GeetikaBatra** [comment]
[test]

**Comment by tuxdna** [comment]
This change is insufficient. Two things are required for adding a table

 * Create an alembic migration - https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/docs/alembic.md and ( optionally http://alembic.zzzcomputing.com/en/latest/tutorial.html#create-a-migration-script )
 * Update models.py to add this new table as a python class ( already done by you )

File: f8a_worker/models.py [hunk 1]
```diff
@@ -407,3 +407,12 @@ class RecommendationFeedback(Base):
     stack_id = Column(String(64), ForeignKey(StackAnalysisRequest.id))
     stack_request = relationship("StackAnalysisRequest",
                                  back_populates="feedback")
+
+
+class OSIORegisteredRepos(Base):
+    __tablename__ = "osio_registered_repo"
+
+    id = Column(Integer, autoincrement=True, primary_key=True)
```
[the remark this answers is not in the document]
Comment (GeetikaBatra) [reply f8a_worker/models.py]: +1

File: f8a_worker/models.py [hunk 2]
```diff
@@ -407,3 +407,12 @@ class RecommendationFeedback(Base):
...
+    github_repo = Column(String(255), nullable=False)
+    github_sha = Column(String(255), nullable=False)
```
[the remark this answers is not in the document]
Comment (GeetikaBatra) [reply f8a_worker/models.py]: sure

File: f8a_worker/models.py [hunk 3]
```diff
@@ -407,3 +407,12 @@ class RecommendationFeedback(Base):
...
+    email_ids = Column(String(255), nullable=False)
```
[the remark this answers is not in the document]
Comment (GeetikaBatra) [reply f8a_worker/models.py]: Will add one. (y)

**Comment by samuzzal-choudhury** [comment]
@tuxdna the alembic migration script is located in `fabric8-analytics-server` and a new version will have to be generated and committed to that repo.

**Comment by GeetikaBatra** [comment]
@tuxdna and @samuzzal-choudhury. So currently, I need to use  https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/docs/alembic.md  or the one in server.

**Comment by msrb** [comment]
> the alembic migration script is located in fabric8-analytics-server

Hmm, isn't it a leftover (?) I don't think we actually run the alembic scripts from server repository. Although I agree that if the tables are meant to be used by just server/gemini, then having them in repositories which own them should be better. If they don't have relationships with other worker tables.

**Comment by samuzzal-choudhury** [comment]
@GeetikaBatra let's go ahead with worker. @msrb we need to remove alembic reference from the server. But, that's some other time.

**Comment by GeetikaBatra** [comment]
@samuzzal-choudhury, sure. Thanks.

**Comment by samuzzal-choudhury** [comment]
@tuxdna my bad. thanks for pointing to the right direction.

File: alembic/versions/967036f90ba1_gemini_model_py.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('osio_registered_repo',
+    sa.Column('git_url', sa.String(length=255), nullable=False),
```
Comment (tuxdna) [inline alembic/versions/967036f90ba1_gemini_model_py.py]: A URL could be more than 255 characters.

File: alembic/versions/967036f90ba1_gemini_model_py.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    sa.Column('git_sha', sa.String(length=255), nullable=False),
+    sa.Column('email_ids', sa.String(length=255), nullable=False),
+    sa.Column('last_scanned_at', sa.DateTime(), nullable=True),
+    sa.PrimaryKeyConstraint('git_sha')
+    )
+    op.create_index(op.f('ix_versions_synced2graph'), 'versions', ['synced2graph'], unique=False)
+    op.drop_index('ix_synced2graph', table_name='versions')
```
Comment (tuxdna) [inline alembic/versions/967036f90ba1_gemini_model_py.py]: This line seems incorrect.

Comment (tuxdna) [reply alembic/versions/967036f90ba1_gemini_model_py.py]: Line 28 and 29 should be removed.

File: alembic/versions/967036f90ba1_gemini_model_py.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_index('ix_synced2graph', 'versions', ['synced2graph'], unique=False)
+    op.drop_index(op.f('ix_versions_synced2graph'), table_name='versions')
```
Comment (tuxdna) [inline alembic/versions/967036f90ba1_gemini_model_py.py]: Linke 35 and 36 should be removed.

File: alembic/versions/967036f90ba1_gemini_model_py.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('osio_registered_repo',
```
Comment (tuxdna) [inline alembic/versions/967036f90ba1_gemini_model_py.py]: One more change here. The table names should be plural. So this should be `osio_registered_repo`.

Comment (tuxdna) [reply alembic/versions/967036f90ba1_gemini_model_py.py]: `osio_registered_repos`

File: f8a_worker/models.py [hunk 8]
```diff
@@ -407,3 +407,12 @@ class RecommendationFeedback(Base):
...
                                  back_populates="feedback")
+
+
+class OSIORegisteredRepos(Base):
+    __tablename__ = "osio_registered_repos"
+
+    git_url = Column(Text, nullable=False, primary_key=True)
+    git_sha = Column(String(255), nullable=False)
+    email_ids = Column(String(255), nullable=False)
+    last_scanned_at = Column(DateTime)
```
Comment (samuzzal-choudhury) [inline f8a_worker/models.py]: @GeetikaBatra  this column should have been timestamp.

File: alembic/versions/967036f90ba1_gemini_model_py.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+import sqlalchemy as sa
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('osio_registered_repos',
+                    sa.Column('git_url', sa.String(length=255), nullable=False),
+                    sa.Column('git_sha', sa.Text, nullable=False),
+                    sa.Column('email_ids', sa.String(length=255), nullable=False),
+                    sa.Column('last_scanned_at', sa.DateTime(), nullable=True),
```
Comment (samuzzal-choudhury) [inline alembic/versions/967036f90ba1_gemini_model_py.py]: This column should have been timestamp

Comment (GeetikaBatra) [reply alembic/versions/967036f90ba1_gemini_model_py.py]: @samuzzal-choudhury, https://github.com/fabric8-analytics/fabric8-gemini-server/pull/1#discussion-diff-171189910R93. I have handled it here.

**Comment by GeetikaBatra** [comment]
@tuxdna, I have made the changes to the Alembic script and made the table name plural, PTAL.

**Comment by samuzzal-choudhury** [comment]
@GeetikaBatra LGTM. Merging. @tuxdna we will address anything pending from your side in a subsequent PR.

**Comment by GeetikaBatra** [comment]
Thanks @samuzzal-choudhury

**Pull request merged**